### ISSUE TITLE
Option to quickly switch to another keyboard

### ISSUE THREAD

**Issue by boikyeet** [8VIM/8VIM#437]
Is it me or there's no way of quickly switching to a different keyboard (switching back to Gboard for example) from the keyboard itself?

This is absolutely necessary in any mobile keyboard

**Comment by MaethorNaur** [comment]
You can quickly switch from 8vim to gboard by setting gboard as the emoji keyboard. However, switch gboard to 8vim you will have to press on the shortcut on the down right of your screen (keyboards selection)

**Comment by ManDay** [comment]
There is https://github.com/8VIM/8VIM/issues/308. Switching to another keyboard should be as easy as calling to [`showInputMethodPicker()`](https://developer.android.com/reference/android/view/inputmethod/InputMethodManager#showInputMethodPicker()). I'd also like a *quick* method, like a gesture, until issue 308 is resolved, as switching to Gboard is currently my only way of getting a different layout (cyrillic in my case) and I have the sidebar disabled.

**Comment by MaethorNaur** [comment]
Having a quick gesture to switch is doable as it has a specific event code. Just need to agree upon the movement.
#308 should be straight forward

**Comment by EricDru** [comment]
I use the emoji button to launch voice input.
Keyboard switcher apps are common. I use one from Kunzisoft. It can run from a notification. ATM I have a notification from Android System that does the same. Not sure how.

**Comment by martin-braun** [comment]
@MaethorNaur @EricDru I was fine with the emoji keyboard shortcut for such a long while. I used it not for voice input, but for getting the keyboard of my password manager in front of me. This allows me to load and paste my credentials without leaving the current app. Now I installed FUTO Voice, an open source voice input that processes all your voice on the device only and I'm stuck. Within 8Vim I need permanent access to two keyboards now.

A simple fix would be to show the keyboard select dialog when long-pressing the smiley. This would certainly help for the time being.

A much more configurable system that results in multiple shortcuts would be anticipated, but I'm not 100% sure how this could work out in the restricted UI, it's already very dense. Maybe the Ctrl/fn toggles could change the functionality of the smiley key, so that we could have 3 keyboards + long press for the keyboard select dialog? That would be a game changer for me, once again.

**Comment by ygrek** [comment]
ftr on android 14 I have keyboard switcher builtin, see screenshot:
![Screenshot_20240817-143500](https://github.com/user-attachments/assets/3e7d2869-22fa-4828-8653-4edc0e62c4b0)

(and now I have a question - how to unassign emoji button to the emoji function again)

**Comment by MaethorNaur** [comment]
@ygrek what do you mean? in a custom layout?

**Comment by MaethorNaur** [comment]
@martin-braun as ygrek said, you should have the keyboard selection from Android. However, that selection might not appear when the keyboard is popped up. So yes adding an icon/movement to show that the selection window could be nice in that case.

**Comment by martin-braun** [comment]
@MaethorNaur @ygrek False. One of my devices has hardware keys, so software buttons are not needed. Thus there is no software button to change the keyboard:

![session-attachment-2024-10-29-023333](https://github.com/user-attachments/assets/12f1fed3-4cf1-4aa1-a73b-da8b10d18f08)

This screenshot is not cropped. Hiding the keyboard also doesn't reveal any button to change the keyboard.

**Comment by MaethorNaur** [comment]
On which version of Android are you running on? (just to know for testing)

**Comment by martin-braun** [comment]
@MaethorNaur My bad, this is still Android 13. If we are certain that the button will be available (even when using hardware buttons and thus having no software buttons), I wouldn't mind further. I got used to this limitation anyways, but I would prefer it to be fixed if the Android software button isn't ensured on all devices.

(I still advocate for the option to long-press the smiley for the menu or a third keyboard option.)

**Comment by MaethorNaur** [comment]
Definitly. It's not a complicated task. I'll probably add a new event so custom layouts can have access to that functionality (all actions like pressing an icon on the side or make a movement produce an event, so anyone can "customise" as they went)

**Comment by Dani3I** [comment]
Also, for some stupid reason, when accessibility button is enabled on android, then it appears in the place of keyboard switch button and thus keyboard switch button "disappears" (instead of going into accessibility settings or on the other side)
